src/main/kotlin/org/jetbrains/kotlinx/jupyter/serializationUtils.kt

```diff
@@ -0,0 +1,377 @@
+package org.jetbrains.kotlinx.jupyter
+
+import org.jetbrains.kotlinx.jupyter.api.VariableState
+import org.jetbrains.kotlinx.jupyter.compiler.util.SerializedVariablesState
+import java.lang.reflect.Field
+import kotlin.reflect.KClass
+import kotlin.reflect.KProperty
+import kotlin.reflect.KProperty1
+import kotlin.reflect.KTypeParameter
+import kotlin.reflect.full.declaredMemberProperties
+import kotlin.reflect.full.isSubclassOf
+import kotlin.reflect.jvm.isAccessible
+
+typealias FieldDescriptor = Map<String, SerializedVariablesState?>
+typealias MutableFieldDescriptor = MutableMap<String, SerializedVariablesState?>
+typealias PropertiesData = Collection<KProperty1<out Any, *>>
+
+data class ProcessedSerializedVarsState(
```

Comment:
Should it really be `data`? I mean, do you use it somewhere where `equals` and `hashCode` are needed, i.e. in map keys or in sets? If no, just make it a simple class and remove unnecessary overrides

Reply:
Fair enough